### PR TITLE
Added RELEASING instructions.

### PR DESCRIPTION
**Pull request by dblock**
Signed-off-by: dblock <dblock@amazon.com>

### Description

Step-by-step guide to making a release.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -32,3 +32,11 @@ Repositories create consistent release labels, such as `v1.0.0`, `v1.1.0` and `v
 ## Releasing
 
 The release process is standard across repositories in this org and is run by a release manager volunteering from amongst [MAINTAINERS](MAINTAINERS.md).
+
+1. Create a tag, e.g. `v2.1.0`, and push it to the GitHub repo.
+2. The [opensearch-java-maven-sign-and-release/](https://build.ci.opensearch.org/job/opensearch-java-maven-sign-and-release/) will be automatically kicked off.
+3. Login to [AWS OSS Sonatype](https://aws.oss.sonatype.org/#stagingRepositories), and locate the staged release.
```
Comment (dblock) [inline RELEASING.md:38]: @gaiksaya the @opensearch-project/engineering-effectiveness team uses an account for this with the right permissions - is the username public? can we put it here?

Comment (gaiksaya) [reply RELEASING.md:38]: I do not believe its public. So no

File: RELEASING.md [hunk 2]
```diff
@@ -32,3 +32,11 @@ Repositories create consistent release labels, such as `v1.0.0`, `v1.1.0` and `v
...
+4. `Close` the staged repository, and ensure all checks pass.
+5. `Release` the staged respository.
+6. [Create a release on GitHub](https://github.com/opensearch-project/opensearch-java/releases/new) with release notes.
+7. Increment `systemProp.version` in [gradle.properties](gradle.properties), commit and push.
```
[the remark this answers is not in the document]
Comment (dblock) [reply RELEASING.md]: Updated to say patch.

File: RELEASING.md [hunk 3]
```diff
@@ -32,3 +32,11 @@ Repositories create consistent release labels, such as `v1.0.0`, `v1.1.0` and `v
 ## Releasing
 
 The release process is standard across repositories in this org and is run by a release manager volunteering from amongst [MAINTAINERS](MAINTAINERS.md).
+
+1. Create a tag, e.g. `v2.1.0`, and push it to the GitHub repo.
+2. The [opensearch-java-maven-sign-and-release/](https://build.ci.opensearch.org/job/opensearch-java-maven-sign-and-release/) will be automatically kicked off.
+3. Login to [AWS OSS Sonatype](https://aws.oss.sonatype.org/#stagingRepositories), and locate the staged release.
+4. `Close` the staged repository, and ensure all checks pass.
+5. `Release` the staged respository.
```
Comment (saratvemulapalli) [inline RELEASING.md:40]: Do these steps really help the community ? 
If its internal, I would not have them listed down.

Comment (dblock) [reply RELEASING.md:40]: Anyone with access to sonatype could do this, and it should really be the maintainers of the project. I think the right answer is to also automate this part as well if we're going to continue using this shared account.

Comment (gaiksaya) [reply RELEASING.md:40]: See https://github.com/opensearch-project/opensearch-build/issues/2398 for details. Maybe add it to readme till we resolve it?

Comment (saratvemulapalli) [reply RELEASING.md:40]: Makes sense.

**Review by saratvemulapalli — APPROVED** [review]

**Review by VachaShah — APPROVED** [review]

**Pull request merged**